src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -320,7 +320,15 @@ var setupSearchButtons = () => {
 async function checkPageExistsAndRedirect(event) {
   // ensure we don't follow the initial link
   event.preventDefault();
-  let currentFilePath = `${DOCUMENTATION_OPTIONS.pagename}.html`;
+  let currentFilePath;
+  if (DOCUMENTATION_OPTIONS.BUILDER == "dirhtml") {
+    currentFilePath =
+      DOCUMENTATION_OPTIONS.pagename == "index"
+        ? `/`
+        : `${DOCUMENTATION_OPTIONS.pagename}/`;
+  } else {
+    currentFilePath = `${DOCUMENTATION_OPTIONS.pagename}.html`;
+  }
```

Comment:
is ti worth pulling this into a (private) util function even local to the file ?

Reply:
much DRY such done also rebased :P